types/coin.go

```diff
@@ -226,6 +243,37 @@ func (coins Coins) safePlus(coinsB Coins) Coins {
 	}
 }
 
+// SubCoinByDenom subtracts a single coin from another coin in a set of coins by
+// denomination. If the coins are empty, the coin is added to the empty set. If
+// the resulting coin is zero, it is removed.
+func (coins Coins) SubCoinByDenom(other Coin) Coins {
+	if len(coins) == 0 {
+		return Coins{other}
+	}
+
+	for i, coin := range coins {
+		if coin.Denom == other.Denom {
+			otherNeg := Coin{other.Denom, other.Amount.Neg()}
```

Comment:
```suggestion
			otherNeg := sdk.NewCoin(other.Denom, other.Amount.Neg())
```

Reply:
We're already in the coin package/implementation so I don't think there is a need to do this. See all the other methods as they also use the `Coin` literal.

Reply:
Will resolve this convo, but reopen if you have any objections 😄